pokedex/storage.lua

```diff
@@ -231,35 +239,30 @@ function M.add(pokemon)
 
 	local id = get_id(pokemon)
 	pokemon.id = id
-	profiles.update(profiles.get_active_slot(), counters)
+
+	player_pokemon[id] = pokemon
 	if M.is_party_full() then
 		pokemon.location = LOCATION_PC
 		pokemon_by_location.pc[id] = true
+		broadcast.send(messages.PC_UPDATED)
 	else
 		pokemon.location = LOCATION_PARTY
 		pokemon.slot = #M.list_of_ids_in_party() + 1
 		pokemon_by_location.party[id] = true
+		broadcast.send(messages.PARTY_UPDATED, {party=get_party()})
 	end
-	player_pokemon[id] = pokemon
 
-	profiles.set_party(get_party())
-	M.save()
-	profiles.save()
+	broadcast.send(messages.SAVE_POKEMON)
+	broadcast.send(messages.COUNTERS_UPDATED, {counters=counters})
 end
 
-
-function M.save()
-	if profiles.get_active_slot() then
-		local profile = profiles.get_active_file_name()
-		defsave.set(profile, "storage_data", storage_data)
-		defsave.save(profile)
-	end
+function M.get_data()
+	return storage_data
 end
 
-function M.upgrade_data(file_name, storage_data)
+local function upgrade_data(file_name, storage_data)
 	local version = storage_data and storage_data.version or 1
 
-	local LATEST_VERSION = 2
 	local needs_upgrade = version ~= LATEST_VERSION
 	
 	if needs_upgrade then
```

Comment:
Inside the below for loop, there's a comment about updating LATEST_VERSION - I would clarify that comment to indicate that the variable is at the top of the file.